pkg/spanconfig/spanconfigtestutils/utils.go

```
@@ -51,12 +55,47 @@ func ParseSpan(t *testing.T, sp string) roachpb.Span {
 	}
 }
 
+// parseSystemTarget is a helepr function that constructs a
+// spanconfig.SystemTarget from a string of the form {source=<id>,target=<id>}
+func parseSystemTarget(t *testing.T, systemTarget string) spanconfig.SystemTarget {
+	if !systemTargetRe.MatchString(systemTarget) {
+		t.Fatalf("expected %s to match system target regex", systemTargetRe)
+	}
+	matches := systemTargetRe.FindStringSubmatch(systemTarget)
+	var sourceID, targetID int
+	var err error
+	if matches[1] == "system" {
+		sourceID = 1
+	} else {
+		sourceID, err = strconv.Atoi(matches[1])
+		require.NoError(t, err)
+	}
+	if matches[2] == "system" {
+		targetID = 1
+	} else {
+		targetID, err = strconv.Atoi(matches[2])
+		require.NoError(t, err)
+	}
+	target, err := spanconfig.MakeSystemTarget(
+		roachpb.MakeTenantID(uint64(sourceID)),
+		roachpb.MakeTenantID(uint64(targetID)),
+	)
+	require.NoError(t, err)
+	return target
+}
+
 // ParseTarget is a helper function that constructs a spanconfig.Target from a
 // string that conforms to spanRe.
-// TODO(arul): Once we have system targets, we'll want to parse them here too
-// instead of just calling ParseSpan here.
 func ParseTarget(t *testing.T, target string) spanconfig.Target {
-	return spanconfig.MakeTargetFromSpan(ParseSpan(t, target))
+	switch {
+	case spanRe.MatchString(target):
+		return spanconfig.MakeTargetFromSpan(ParseSpan(t, target))
+	case systemTargetRe.MatchString(target):
+		return spanconfig.MakeTargetFromSystemTarget(parseSystemTarget(t, target))
+	default:
+		t.Fatalf("expected %s to match span or system target regex", target)
+	}
+	return spanconfig.Target{}
```

Could panic instead of this confusing return.